yarn-project/stdlib/src/snapshots/download.ts

```diff
@@ -50,10 +53,39 @@ export function makeSnapshotPaths(baseDir: string): SnapshotDataUrls {
   return fromEntries(SnapshotDataKeys.map(key => [key, `${baseDir}/${key}.db`]));
 }
 
+function maybeGunzip(data: Buffer): Buffer {
+  // detect gzip magic number 1F 8B
+  if (data.length >= 2 && data[0] === 0x1f && data[1] === 0x8b) {
```

Comment:
Not familiar with R2, but is there a header or tag we can rely on for detecting whether we need to decompress?